scripts/comfyui.py

```diff
@@ -1,71 +1,122 @@
+
         with gr.Row():
-            queue_front = gr.Checkbox(label='Queue front', elem_id='sd-comfyui-webui-queue_front', value=True)
-            output_node_label = gr.Dropdown(label='Workflow type', choices=['postprocess'], value='postprocess')
+            queue_front = gr.Checkbox(
+                label='Queue front',
+                elem_id=self.elem_id('queue_front'),
+                value=True,
+            )
+            workflow_types = external_code.get_workflow_types(xxx2img)
+            workflow_type_ids = {
+                workflow_type.display_name: workflow_type.get_ids(xxx2img)[0]
+                for workflow_type in workflow_types
+            }
+            workflow_display_name = gr.Dropdown(
+                label='Edit workflow type',
+                choices=[workflow_type.display_name for workflow_type in workflow_types],
+                value=workflow_types[0].display_name,
+                elem_id=self.elem_id('displayed_workflow_type'),
+            )
+            current_workflow_type_id = gr.Text(
+                value=workflow_type_ids[workflow_display_name.value],
+                visible=False,
+                interactive=False,
+            )
+            workflow_display_name.change(
+                fn=workflow_type_ids.get,
+                inputs=[workflow_display_name],
+                outputs=[current_workflow_type_id],
+            )
+            current_workflow_type_id.change(
+                fn=None,
+                _js='changeDisplayedWorkflowType',
+                inputs=[current_workflow_type_id],
+            )
+
         with gr.Row():
-            gr.HTML(value=f"""
-                <iframe src="{webui_settings.get_comfyui_client_url()}" id="comfyui_postprocess_{xxx2img}" class="comfyui-embedded-widget" style="width:100%; height:500px;"></iframe>
-            """)
+            gr.HTML(value=self.get_iframes_html(is_img2img))
+
         with gr.Row():
-            refresh_button = gr.Button(value=f'{ui.refresh_symbol} Reload ComfyUI interface (client side)', elem_id='sd-comfyui-webui-refresh_button')
+            refresh_button = gr.Button(value=f'{ui.refresh_symbol} Reload ComfyUI interface (client side)', elem_id=self.elem_id('refresh_button'))
             refresh_button.click(
                 fn=None,
                 _js='reloadComfyuiIFrames'
             )
-        return queue_front, output_node_label
 
-    def show(self, is_img2img):
-        return scripts.AlwaysVisible
+        return queue_front,
+
+    def get_iframes_html(self, is_img2img: bool) -> str:
+        comfyui_client_url = settings.get_comfyui_client_url()
+
+        iframes = []
+        first_loop = True
+        for workflow_type_id in external_code.get_workflow_type_ids(self.get_xxx2img_str(is_img2img)):
+            html_classes = ['comfyui-embedded-widget']
+            if first_loop:
+                first_loop = False
+                html_classes.append('comfyui-embedded-widget-display')

```

Comment:
let me know if I'm wrong, but I think it would be better to use the default element that was set in the `gr.Dropdown():` as the first displayed value here

Reply:
Sure, I can refact the condition. It's the first one in the dropdown so it has the same behavior.